calnex/api/api_test.go

```diff
@@ -760,3 +760,19 @@ func TestPushLicenseError(t *testing.T) {
 
 	require.Equal(t, originalLicense, uploadedLicense)
 }
+
+func TestReboot(t *testing.T) {
+
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter,
```

Comment:
It's a little more complex than this. 
Reboot API first checks the status (if measurement is running):
https://github.com/facebook/time/blob/main/calnex/api/api.go#L824-L838

Thats why you need to check the status first (similar to what is happening in TestGet):
https://github.com/facebook/time/blob/main/calnex/api/api_test.go#L515-L523

As you see just successful "reboot" is already tested in https://github.com/facebook/time/blob/main/calnex/api/api_test.go#L539-L540

Maybe we can use an opportunity to test different scenarios, for example when error is returned by the device.

Reply:
```
func TestReboot(t *testing.T) {

	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter,
		r *http.Request) {
		fmt.Fprintln(w, "Calnex device will now reboot.")
	}))
	defer ts.Close()

	parsed, _ := url.Parse(ts.URL)
	calnexAPI := NewAPI(parsed.Host, true)
	calnexAPI.Client = ts.Client()

	status, err := calnexAPI.FetchStatus()
	if err != nil {
		require.NoError(t, err, "Fetching the status returned an error")
	}

	if status.MeasurementActive {
		if err = calnexAPI.StopMeasure(); err != nil {
			require.NoError(t, err, "Measurement is not active")
		}
	}
	calnexAPI.get(rebootURL)
	err = calnexAPI.Reboot()

	require.NoError(t, err, "reboot returned an error")
}
```
I added the status checks. Does this look ok? @leoleovich

Reply:
You don't need to fetch status - it's fetched automatically in api.Reboot(). What you need to do is to supply the response via http server (pretty much like in TestGet I linked above). What you need to do (to make it different from TestGet is to test "fail" Result case when reboot api call failed and check proper error is returned (t.ErrorIs).
Also don't forget to squash commits. Currently you have 3 commits. Make sure your PR is only 1.

Reply:
I created a  test case for failed reboot I added response code 500 for the http failed server. I have added these lines alongwith the code 
```bash
var ErrorRebootFailed = fmt.Errorf("failed to reboot Calnex device")
	ts = httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		http.Error(w, "reboot failed", http.StatusInternalServerError)
	}))
	defer ts.Close()
	parsed, _ = url.Parse(ts.URL)
	calnexAPI = NewAPI(parsed.Host, true)
	calnexAPI.Client = ts.Client()

	err = calnexAPI.Reboot()
	require.ErrorIs(t, err, ErrorRebootFailed, "failed to reboot Calnex device")
```

Reply:
That is actually not bad!

Reply:
I am adding this to the code and squashing the commits

Reply:
Added the LOC